unstructured_ingest/v2/processes/connectors/vectara.py

```diff
@@ -0,0 +1,323 @@
+                }
+            }
+            create_corpus_response = await self._request(endpoint="create-corpus", data=data)
+            self.connection_config.corpus_id = create_corpus_response.get("corpusId")
+
+    @requires_dependencies(["httpx"], extras="vectara")
+    async def _request(
+        self,
+        endpoint: str,
+        http_method: str = "POST",
+        params: Mapping[str, Any] = None,
+        data: Mapping[str, Any] = None,
+    ):
+        import httpx
+
+        url = f"{BASE_URL}/{endpoint}"
+
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+            "Authorization": f"Bearer {await self.jwt_token}",
+            "customer-id": self.connection_config.customer_id,
+            "X-source": "unstructured",
+        }
+
+        async with httpx.AsyncClient() as client:
+            response = await client.request(
+                method=http_method, url=url, headers=headers, params=params, json=data
+            )
+            response.raise_for_status()
+            return response.json()
+
+    async def _delete_doc(self, doc_id: str) -> None:
+        """
+        Delete a document from the Vectara corpus.
+
+        Args:
+            url (str): URL of the page to delete.
+            doc_id (str): ID of the document to delete.
+        """
+        body = {
+            "customer_id": self.connection_config.customer_id,
+            "corpus_id": self.connection_config.corpus_id,
+            "document_id": doc_id,
+        }
+        await self._request(endpoint="delete-doc", data=body)
+
+    async def _index_document(self, document: Dict[str, Any]) -> None:
+        """
+        Index a document (by uploading it to the Vectara corpus) from the document dictionary
+        """
+        body = {
+            "customer_id": self.connection_config.customer_id,
+            "corpus_id": self.connection_config.corpus_id,
+            "document": document,
+        }
+
+        logger.debug(
+            f"Indexing document {document['documentId']}"
+            + " to corpus ID {self.connection_config.corpus_id}"
+        )
+
+        try:
+            result = await self._request(endpoint="index", data=body, http_method="POST")
+        except Exception as e:
+            logger.info(f"exception {e} while indexing document {document['documentId']}")
+            return
+
+        if (
+            "status" in result
+            and result["status"]
+            and (
+                "ALREADY_EXISTS" in result["status"]["code"]
+                or (
+                    "CONFLICT" in result["status"]["code"]
+                    and "Indexing doesn't support updating documents"
+                    in result["status"]["statusDetail"]
+                )
+            )
+        ):
+            logger.info(f"document {document['documentId']} already exists, re-indexing")
+            await self._delete_doc(document["documentId"])
```

Comment:
Is there no upsert or overwrite support from vectara? I'm hesitant to explicitly delete anything.